src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs

```diff
@@ -183,14 +183,20 @@ private void LogProductionResult(Task<Block?> t, Block currentBestBlock, UInt256
             {
                 bool supportsBlobs = _blockProducer.SupportsBlobs;
                 int blobs = 0;
+                int blobTx = 0;
                 if (supportsBlobs)
                 {
                     foreach (Transaction tx in block.Transactions)
                     {
-                        blobs += tx.BlobVersionedHashes?.Length ?? 0;
+                        int blobCount = tx.GetBlobCount();
+                        if (blobCount > 0)
+                        {
+                            blobs += blobCount;
+                            blobTx++;
+                        }
```

Comment:
Move this code to helper method?

Reply:
Is only used in one place though; which is this log 🤔